### PR TITLE
hbacsvcgroup: Fix member management idempotence issues.

### PR DESCRIPTION
**Pull request by rjeffman**
The hbacsvc members of hbacsvcgroup must be compared in a case
insensitive manner. This patch fixes comparation of member parameters
against existing members by converting parameters to lowercase, as it
is how the hbacsvc members are stored for hbacsvcgroups.

Also, there were some cases where a change with an empty set of members
was issued to IPA API, leading to a result of `changed: yes` when
`changed: no` was expected. The fix involved a refactoring of the
hbacsvcgroup member management code.

This PR should not be merged before PR #683.

> Note: commit ca0aed54b0ab9431a8dc2653ccada66f967e7a3e is part of #683.

### CODE REVIEW

File: plugins/modules/ipahbacsvcgroup.py [hunk 1]
```diff
@@ -243,32 +246,21 @@ def main():
                     if not compare_args_ipa(ansible_module, member_args,
                                             res_find):
                         # Generate addition and removal lists
-                        hbacsvc_add, hbacsvc_del = gen_add_del_lists(
-                            hbacsvc, res_find.get("member_hbacsvc"))
+                        if hbacsvc is not None:
+                            hbacsvc_add, hbacsvc_del = gen_add_del_lists(
+                                list(set(hbacsvc)),
```
Comment (t-woerner) [inline plugins/modules/ipahbacsvcgroup.py:251]: Why is this here?

Comment (rjeffman) [reply plugins/modules/ipahbacsvcgroup.py]: The original goal was to remove possible duplicates created by the modified list.
As we discussed offline, it is better to not filter the input in this case. Fixed.

**Review by t-woerner — APPROVED** [review]
LGTM
